### PR TITLE
Run phpunit even when phpcs fails

### PR DESCRIPTION
**Pull request by noahtallen**
## Description
Over the past several hours, the phpcs test has been failing on master. This was eventually fixed, but it meant that several PRs touching PHP code were merged in the meantime. Some of these PRs broke phpunit tests. However, the phpunit failure was not obvious because when phpcs fails, the entire step stops. Therefore phpunit does not run when phpcs fails.

The solution is to run phpunit even when phpcs fails.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -83,8 +83,17 @@ jobs:
         chmod -R 767 ./ # TODO: Possibly integrate in wp-env
         npm run wp-env start
 
-    - name: Running the tests
-      run: npm run test-php && npm run test-unit-php-multisite
```
Comment (noahtallen) [inline .github/workflows/unit-test.yml:87]: `test-php` just runs `lint-php && test-unit-php`, so we are just splitting it out into its own steps

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -83,8 +83,17 @@ jobs:
...
+    - name: Running lint check
+      run: npm run lint-php
+
+    - name: Running single site unit tests
+      run: npm run test-unit-php
+      if: ${{ success() || failure() }}
```
Comment (noahtallen) [inline .github/workflows/unit-test.yml:91]: could also use `always()` here, but this should basically mean "run this all the time unless the run was cancelled"

**Comment by noahtallen** [comment]
As you can see, it now reports errors for each of the suites, even if only one of them failed:

Screen Shot 2020-11-16 at 7 01 39 PM

Previously, that would have skipped over the unit tests and just run the lint check.

if folks like this change, I'll remove the test commit and merge!

**Comment by noahtallen** [comment]
And here's an example of the lint check failing and the phpunit tests passing:

Screen Shot 2020-11-16 at 7 45 59 PM

**Comment by noisysocks** [comment]
Sounds reasonable to me. Do we have a similar problem with JavaScript unit tests?

**Comment by tellthemachines** [comment]
> Do we have a similar problem with JavaScript unit tests?

Looks like we have the [same setup](https://github.com/WordPress/gutenberg/blob/master/package.json#L245) there. Probably worth changing just for the sake of consistency it nothing else 😄

**Comment by TimothyBJacobs** [comment]
Huge +1 to this.

**Comment by noahtallen** [comment]
> Looks like we have the same setup there. Probably worth changing just for the sake of consistency it nothing else 😄

I checked the workflow files, and it looks like we just use the `test-unit-js` command directly with no lint check

https://github.com/WordPress/gutenberg/blob/a57cc161a594e00bb12a83d6f2e4921bad70389f/.github/workflows/unit-test.yml#L48

https://github.com/WordPress/gutenberg/blob/a57cc161a594e00bb12a83d6f2e4921bad70389f/package.json#L252

**Review by Addison-Stavlo — APPROVED** [review]
Definitely a well needed change.

This makes sense to me and seems to be running the 3 stages well!

**Review by vindl — APPROVED** [review]
LGTM :shipit:

**Pull request merged**